lib/AST/Decl.cpp

```diff
@@ -1165,7 +1165,13 @@ static bool isPolymorphic(const AbstractStorageDecl *storage) {
 
   case DeclKind::Class:
     // Final properties can always be direct, even in classes.
-    return !storage->isFinal();
+    if (storage->isFinal())
+      return false;
+    // Extension properties are statically dispatched, unless they're @objc.
+    if (storage->getDeclContext()->isExtensionContext()
+        && !storage->isObjC())
```

Comment:
I'd rather just be implicitly marking extension declarations `final` than put special cases for them all over the place. This might be the *right* place to put one, but it's not the only one.